### ISSUE TITLE
undefined reference to `yasem::InterceptorManager::onMetadataChanged(NetworkReply*)'

### ISSUE THREAD

**Issue by afterburn420** [mvasilchuk/yasem#11]
When trying to compile in windows using qtcreator I keep getting this error

C:\yasem\wj\yasem\plugins\yasem-web-browser\debug\moc_interceptormanager.cpp:84: error: undefined reference to `yasem::InterceptorManager::onMetadataChanged(NetworkReply*)'

C:\yasem\wj\yasem\plugins\yasem-web-browser\debug\moc_interceptormanager.o:-1: In function `ZN5yasem18InterceptorManager18qt_static_metacallEP7QObjectN11QMetaObject4CallEiPPv':

which is in reference to this line in moc.interceptormanager.cpp>>>
        case 2: _t->onMetadataChanged((*reinterpret_cast< NetworkReply*(*)>(_a[1]))); break;

It happens no matter what version of qt I use (5.2.1, 5.3, or 5.4) or version of ffmpeg/libav etc.

I also only get the 1.a and 1.dll version of yasem-qtav-mediaplayer even though all other plugins are 0.a and 0.dll (no difference if built as release version or debug version).

I have to comment out that line to finish compile and then it will not accept yasem-qtav-mediaplayer1.dll and says media player plugin not found

**Comment by afterburn420** [comment]
The only thing that I have thought of so far is that maybe there is an incompatibility with the windows versions of the qt modules. Either that or some file became corrupt in my toolchain or modules.

**Comment by mvasilchuk** [comment]
I'll try to compile yasem on Windows to check what may be wrong.

**Comment by mvasilchuk** [comment]
I've fixed that bug (I forgot to make an implementation of that method but don't know why it works on Linux) and also changed yasem-qtav-mediaplayer's version back to 0.1.0 to be similar to other plugins (see VERSION field in yasem-qtav-mediaplayer.pro).
But anyways you won't be able to run it on Windows because you need first copy libav/ffmpeg binaries into plugins directory and also some other libs from mingw you used to build the app. Use Dependency Walker (http://dependencywalker.com/) on yasem-qtav-mediaplayer0.dll to see what libraries you need.

**Comment by afterburn420** [comment]
now I am getting a lot of errors related to QtAv::OutputSet

I noticed that there are includes that are not in the project for example -->
#include "QtAV/private/VideoRenderer_p.h"
#include "QtAV/private/prepost.h"

The compiler output has many lines like this-->
./..\.obj\Qt5AVWidgetsd_win_x86\OpenGLWidgetRenderer.o:OpenGLWidgetRenderer.cpp:(.rdata$_ZTVN4QtAV20OpenGLWidgetRendererE[__ZTVN4QtAV20OpenGLWidgetRendererE]+0x134): undefined reference to `QtAV::AVOutput::onAddOutputSet(QtAV::OutputSet*)'

**Comment by mvasilchuk** [comment]
Try to get the last libav library (I used libav for compiling, not ffmpeg) and clean app output directory (or just delete it). If it doesn't help remove yasem-qtav-plugin directory and pull it from git again.

**Comment by afterburn420** [comment]
What version are you using - I have v11 and v11.1 - and neither have swresample, only avresample.

When I use them I get an error because of not having libswresample

**Comment by mvasilchuk** [comment]
I used 11.1 (i686) from http://builds.libav.org/windows/release-gpl/

**Comment by afterburn420** [comment]
Those are the ones that I have - from the link you gave me in another post.

They give me an error because they do not include libswresample.

avcompat.h is insisting on libswresample instead of using libavresample

**Comment by afterburn420** [comment]
Even if I cheat the problem by including swresample to libav or avresample to ffmpeg I get the errors-->
C:\yasem-wj\yasem3\plugins\yasem-qtav-mediaplayer\QtAV\src\output\video\OpenGLWidgetRenderer.cpp:-1: error: undefined reference to `QtAV::AVOutput::onAddOutputSet(QtAV::OutputSet*)'

I get 24 errors all about QtAv::OutputSet  that are among several files including-->
OpenGLWidgetRenderer.cpp, GDIRenderer.cpp, Direct2DRenderer.cpp, moc.GraphicsItemRenderer.cpp, moc.WidgetRenderer.cpp, and moc.GLWidgetRenderer.cpp

**Comment by mvasilchuk** [comment]
Try to update your sources and recompile the project again. And if you get the same errors try to change lines in yasem-qtav-plugin/QtAvMediaPlayer/QtAvMediaPlayer.pro:
win32:CONFIG(release, debug|release): LIBS += -L$$BUILD_DIR/bin -lQt5AV -lQt5AVWidgets
else:win32:CONFIG(debug, debug|release): LIBS += -L$$BUILD_DIR/bin -lQt5AVd -lQt5AVWidgetsd

to 

win32:CONFIG(release, debug|release): LIBS += -L$$BUILD_DIR/bin -lQtAV -lQtAVWidgets
else:win32:CONFIG(debug, debug|release): LIBS += -L$$BUILD_DIR/bin -lQtAVd -lQtAVWidgetsd

(remove "5" from -lQt5AV and -lQt5AVWidgets and theirs debug variants).
It may help you. 
I don't know why yet, but if you use Qt 5.4 compiler makes QtAV libs with names Qt5AV and Qt5AVWidgets, but if you use previous versions of Qt it makes QtAV and QtAVWidgets.

**Comment by afterburn420** [comment]
are you using the mingw or the msvc compiler - and what version of them.

I am using mingw because I can't get my msvc one to find my include and library paths.

**Comment by afterburn420** [comment]
I have noticed that all plugins give the error on loading-->
plugin verification data mismatch

**Comment by afterburn420** [comment]
This is the error that debug gives-->

[DEBUG] Found file: "yasem-qtav-mediaplayer0.dll"
[WARN ] "Cannot load library C:/yasem-wj/build-yasemer4/bin/plugins/yasem-qtav-mediaplayer0.dll: The specified module could not be found."

**Comment by afterburn420** [comment]
here is the debug from yasem-->

[DEBUG] Library paths:  ("C:/Qt/5.3/mingw482_32/plugins", "C:/yasem-wj/build-yasemer8/bin", "C:/yasem-wj/build-yasemer8/bin/libs", "C:/yasem-wj/build-yasemer8/bin/libs/plugins", "")
[DEBUG] Settings directory "C:/Users/afterburn-gamer/AppData/Roaming/yasem"
[DEBUG] Starting application...
[DEBUG] Looking for plugins...
[DEBUG] pluginsToLoad: ("desktop-gui", "dunehd-plugin", "ini-ds", "mag-api", "samsung-smart-tv-plugin", "tray-icon", "webkit-browser", "webserver-browser")
[DEBUG] PluginManager::listPlugins()
[DEBUG] Blacklisted plugins: ()
[DEBUG] Searching for plugins in "C:/yasem-wj/build-yasemer8/bin/plugins"
[DEBUG] Found file: "libyasem-desktop-gui0.a"
[WARN ] "Plugin verification data mismatch in 'C:/yasem-wj/build-yasemer8/bin/plugins/libyasem-desktop-gui0.a'"
[DEBUG] Found file: "libyasem-dunehd-api0.a"
[WARN ] "Plugin verification data mismatch in 'C:/yasem-wj/build-yasemer8/bin/plugins/libyasem-dunehd-api0.a'"
[DEBUG] Found file: "libyasem-ini-datasource0.a"
[WARN ] "Plugin verification data mismatch in 'C:/yasem-wj/build-yasemer8/bin/plugins/libyasem-ini-datasource0.a'"
[DEBUG] Found file: "libyasem-mag-api0.a"
[WARN ] "Plugin verification data mismatch in 'C:/yasem-wj/build-yasemer8/bin/plugins/libyasem-mag-api0.a'"
[DEBUG] Found file: "libyasem-samsung-smarttv-api0.a"
[WARN ] "Plugin verification data mismatch in 'C:/yasem-wj/build-yasemer8/bin/plugins/libyasem-samsung-smarttv-api0.a'"
[DEBUG] Found file: "libyasem-tray-icon0.a"
[WARN ] "Plugin verification data mismatch in 'C:/yasem-wj/build-yasemer8/bin/plugins/libyasem-tray-icon0.a'"
[DEBUG] Found file: "libyasem-web-browser0.a"
[WARN ] "Plugin verification data mismatch in 'C:/yasem-wj/build-yasemer8/bin/plugins/libyasem-web-browser0.a'"
[DEBUG] Found file: "libyasem-web-server0.a"
[WARN ] "Plugin verification data mismatch in 'C:/yasem-wj/build-yasemer8/bin/plugins/libyasem-web-server0.a'"
[DEBUG] Found file: "yasem-desktop-gui0.dll"
[DEBUG] Registering plugin roles...
[DEBUG] Registering plugin dependencies...
[DEBUG] Plugin loaded: "Desktop GUI plugin"
[DEBUG] Registered "config" as profile class ID
[DEBUG] Found STB plugin: "Desktop GUI plugin"
[DEBUG] Found STB submodels: ""
[DEBUG] Found file: "yasem-dunehd-api0.dll"
[DEBUG] Registering plugin roles...
[DEBUG] Registering plugin dependencies...
[DEBUG] Plugin loaded: "DuneHD JS API implementation"
[DEBUG] Registered "dunehd" as profile class ID
[DEBUG] Found STB plugin: "DuneHD JS API implementation"
[DEBUG] Found STB submodels: ""
[DEBUG] Found file: "yasem-ini-datasource0.dll"
[DEBUG] Registering plugin roles...
[DEBUG] Registering plugin dependencies...
[DEBUG] Plugin loaded: "INI datasource provider"
[DEBUG] Found file: "yasem-mag-api0.dll"
[DEBUG] Registering plugin roles...
[DEBUG] Registering plugin dependencies...
[DEBUG] Plugin loaded: "MAG API implementation"
[DEBUG] Registered "mag" as profile class ID
[DEBUG] Found STB plugin: "MAG API implementation"
[DEBUG] Found STB submodels: "MAG100, MAG200, MAG245, MAG250, MAG254, MAG255, MAG260, MAG270, MAG275, AuraHD"
[DEBUG] Found file: "yasem-samsung-smarttv-api0.dll"
[STUB ] yasem::SamsungSmartTvWebPlugin::SamsungSmartTvWebPlugin()
[DEBUG] Registering plugin roles...
[DEBUG] Registering plugin dependencies...
[DEBUG] Plugin loaded: "samsung Smart TV API implementation"
[DEBUG] Registered "samsung" as profile class ID
[DEBUG] Found STB plugin: "samsung Smart TV API implementation"
[DEBUG] Found STB submodels: ""
[DEBUG] Found file: "yasem-tray-icon0.dll"
[DEBUG] Registering plugin roles...
[DEBUG] Registering plugin dependencies...
[DEBUG] Plugin loaded: "Tray icon plugin"
[DEBUG] Found file: "yasem-web-browser0.dll"
[DEBUG] Registering plugin roles...
[DEBUG] Registering plugin dependencies...
[DEBUG] Plugin loaded: "Webkit browser plugin"
[DEBUG] Found file: "yasem-web-server0.dll"
[DEBUG] Registering plugin roles...
[DEBUG] Registering plugin dependencies...
[DEBUG] Plugin loaded: "Web server plugin"
[DEBUG] initPlugins()
[DEBUG]  Plugin "desktop-gui" initialization...
[DEBUG]  Loading dependencies for "desktop-gui" ...
[DEBUG]  Trying to load dependency "Browser plugin"  for "Desktop GUI plugin"
[DEBUG] -- Plugin "webkit-browser" initialization...
[DEBUG] -- Loading dependencies for "webkit-browser" ...
[DEBUG] -- Trying to load dependency "GUI plugin"  for "Webkit browser plugin"
[WARN ] Cyclic dependency in plugin "webkit-browser" with "desktop-gui"
[DEBUG] -- Dependencies for "webkit-browser" have been loaded.
[STUB ] void yasem::WebView::loadFixes()
[STUB ] QString yasem::WebView::loadFix(const QString&)
[STUB ] QString yasem::WebView::loadFix(const QString&)
[STUB ] bool yasem::WebPage::stb(yasem::StbPlugin*)
[STUB ] yasem::InterceptorManager::InterceptorManager(yasem::WebPage*)
[DEBUG] wtf child: (yasem::WebView(0xae9da0, name = "WebView") ) 
[DEBUG] -- Plugin "webkit-browser" initialized
[DEBUG]  Dependency "Browser plugin" initialized
[DEBUG]  Trying to load dependency "Datasource plugin"  for "Desktop GUI plugin"
[DEBUG] -- Plugin "ini-ds" initialization...
[DEBUG] -- Loading dependencies for "ini-ds" ...
[DEBUG] No dependencies for "ini-ds" have been found
[DEBUG] -- Plugin "ini-ds" initialized
[DEBUG]  Dependency "Datasource plugin" initialized
[DEBUG]  Trying to load dependency "Media player plugin"  for "Desktop GUI plugin"
[WARN ] Dependency "Media player plugin" for "desktop-gui" not found!
[WARN ] Plugin "desktop-gui" won't be loaded because of missing dependencies!
[DEBUG]  Plugin "dunehd-plugin" initialization...
[DEBUG]  Loading dependencies for "dunehd-plugin" ...
[DEBUG]  Trying to load dependency "Browser plugin"  for "DuneHD JS API implementation"
[DEBUG]  Dependency "Browser plugin" initialized
[DEBUG]  Dependencies for "dunehd-plugin" have been loaded.
[DEBUG]  Plugin "dunehd-plugin" initialized
[DEBUG]  Plugin "mag-api" initialization...
[DEBUG]  Loading dependencies for "mag-api" ...
[DEBUG]  Trying to load dependency "Datasource plugin"  for "MAG API implementation"
[DEBUG]  Dependency "Datasource plugin" initialized
[DEBUG]  Trying to load dependency "Browser plugin"  for "MAG API implementation"
[DEBUG]  Dependency "Browser plugin" initialized
[DEBUG]  Trying to load dependency "Media player plugin"  for "MAG API implementation"
[WARN ] Dependency "Media player plugin" for "mag-api" not found!
[WARN ] Plugin "mag-api" won't be loaded because of missing dependencies!
[DEBUG]  Plugin "samsung-smart-tv-plugin" initialization...
[DEBUG]  Loading dependencies for "samsung-smart-tv-plugin" ...
[DEBUG]  Trying to load dependency "Browser plugin"  for "samsung Smart TV API implementation"
[DEBUG]  Dependency "Browser plugin" initialized
[DEBUG]  Trying to load dependency "Datasource plugin"  for "samsung Smart TV API implementation"
[DEBUG]  Dependency "Datasource plugin" initialized
[DEBUG]  Trying to load dependency "Media player plugin"  for "samsung Smart TV API implementation"
[WARN ] Dependency "Media player plugin" for "samsung-smart-tv-plugin" not found!
[WARN ] Plugin "samsung-smart-tv-plugin" won't be loaded because of missing dependencies!
[DEBUG]  Plugin "tray-icon" initialization...
[DEBUG]  Loading dependencies for "tray-icon" ...
[DEBUG] No dependencies for "tray-icon" have been found
[DEBUG]  Plugin "tray-icon" initialized
[DEBUG]  Plugin "webserver-browser" initialization...
[DEBUG]  Loading dependencies for "webserver-browser" ...
[DEBUG]  Trying to load dependency "Browser plugin"  for "Web server plugin"
[DEBUG]  Dependency "Browser plugin" initialized
[DEBUG]  Dependencies for "webserver-browser" have been loaded.
[DEBUG] Could not start server: "The bound address is already in use"
[DEBUG]  Plugin "webserver-browser" initialized
[DEBUG] Initialization finished

**Comment by mvasilchuk** [comment]
>> I have noticed that all plugins give the error on loading--> plugin verification data mismatch

It's not a bug, just a little issue. I don't know why but on Windows mingw puts into libs folder both *.dll and *.a files. And *.a files can't be loaded as plugins because they aren't.

**Comment by mvasilchuk** [comment]
>> [WARN ] "Cannot load library C:/yasem-wj/build-yasemer4/bin/plugins/yasem-qtav-mediaplayer0.dll: The specified module could not be found.

Did you put DLLs from libav bin folder into /plugins? Use dependency walker (the link is in one of the prevoius messages ) to get a list of libraries you also need to put there.

**Comment by afterburn420** [comment]
I will see about downloading the libraries that are missing from my computer.

Even with show all files checked there are missing api-ms-win~~.dll files in system 32.

Windows Finder shows that they are not in my computer.

Most likely because I am running win7 64bit - and qtcreator for win only has 32bit mingw support.

**Comment by mvasilchuk** [comment]
you may skip api-ms-*, they aren't required. Afaik they are only needed for some versions of Windows and if they are missing app still works.

**Comment by afterburn420** [comment]
Cool - then there are only a couple to download....thanks

I will give it a try

**Comment by afterburn420** [comment]
I will try to work it out but it is wanting dll files that are not compatible with my system.

For example DCOMP.DLL is a dx11.1 library that is only compatible with windows 8 or newer.

It is for DirectComposition and win 7 does not support those features.

I am not sure if it is my qtcreator or my includes that are causing yasem to want the incompatible libs.

**Comment by mvasilchuk** [comment]
Try to just ignore all DLLs that are not compatible. Actually, I only needed libraries from mingw, Qt (if you want to distribute the app) and DLLs from libav folder.

**Comment by afterburn420** [comment]
I am having no luck.

I have even tried copying every library from all of those folders to both bin and plugins.

Dependency walker only shows DCOMP.DLL, GPSVC.DLL, IESHIMS.DLL and the api-ms files.

dcomp is incompatible - gpsvc and ieshims are 64 bit versions

I still get "media player plugin" not found.

The system that I am using is an AMD_64 (phenom 2 x4)

Do you know what path variables to use to get QTcreators msvc compiler to find lib and include?(maybe I will try that)

**Comment by afterburn420** [comment]
I have noticed that  yasem-qtav-mediaplayer0.dll is not in the "pluginsToLoad:" list.

Is that the way it should be?

**Comment by mvasilchuk** [comment]
"pluginsToLoad" is just a list of modules that should be loaded if available (this list is from config). You may ignore it, because I'm not using it now.

But I've compiled the app and packed it with all libraries that are needed. Try to download and run it https://mega.co.nz/#!8MQRUSjC!wH-i3-zEC3qigwW2o6rzX5MUldOGAsB9a22By_YpRSw

**Comment by afterburn420** [comment]
The version you compiled is working for me.

I noticed that there were some extra codec dlls that are not in my compiles.

I will see about getting them and find out if it helps me compile.

Thank you very much - you have been very kind and patient with me.

**Comment by mvasilchuk** [comment]
You may just get all my dll's from /bin foder and copy them to your compiled binaries.

**Comment by afterburn420** [comment]
Yes - I am going to try that first.

**Comment by afterburn420** [comment]
I was mistaken - all of those were in my libav.

I will look closer into why it is only compiling on my system with qt5.3.

I'll try and find out why qt5.2.1 and qt5.4 are giving errors during compile.

What versions of qt and mingw are you using - example..qt5.4 mingw4.9.1?

**Comment by afterburn420** [comment]
Also on the portal that I am using videoclub works great but live tv channels do not work.

The live tv channels work with my mag 254 receiver.

If you want to test it the portal is http://1.iptvprivateserver.tv - it is on a free trial so it will work for you

**Comment by mvasilchuk** [comment]
I used  Qt 5.4 and mingw 4.9.1 to compile the app I've sent you. And I've just tested your portal and sometimes live channels worked but sometimes they didn't. I'll try to fix this issue if it's not a QtAV bug.

**Comment by afterburn420** [comment]
after a fresh install of qtcreator 5.4 - it now compiles but mediaplayer still not found.

Is there anything besides libav that is required for depends to make it work.

here is my depends list-->
Checking for avutil... yes
Checking for avcodec... yes
Checking for avformat... yes
Checking for swscale... yes
Project MESSAGE: checking for optional features...
Checking for swresample... no
Checking for avresample... yes
Checking for avdevice... yes
Checking for gl... yes
Checking for avfilter... no
Checking for sse4_1... yes
Checking for openal... no
Checking for portaudio... no
Checking for direct2d... yes
Checking for gdiplus... yes
Checking for dxva... yes
Checking for libass... no

**Comment by afterburn420** [comment]
After using git to download the new version with degug and release fix it is working.

I used the ffmpeg for qtav and the video is working better.(filename QtAV-depends)

On live channels I just have to exit and reenter tv menu to get a new channel to play (after changing the channel first)

Now I will check to see if directx is working with the video.

Thanks for the help I owe you - if you need something let me know....and keep up the good work.

P.S. when the server that I am using goes fully online there will be thousands of people likely wanting to use your software (stbemu and yasem) - there are many using stbemu with bluestacks android emu now.

**Comment by mvasilchuk** [comment]
It would be great if you test it.

BTW, my current app GUI isn't good, so if you need to modify it you may find app web-pages in plugins/yasem-web-gui-plugin/gui

**Comment by mvasilchuk** [comment]
Try a new version. I've fixed that bug.

**Comment by afterburn420** [comment]
Thanks - I'll "git" it now and try it

**Comment by afterburn420** [comment]
With the new update I get a compile error -->
C:\yasem-fix\build-yasem-fix2\bin\libQt5AV1.a(d000960.o):-1: error: multiple definition of `QtAV::AVPlayer::mediaStatusChanged(QtAV::MediaStatus)'

moc_AVPlayer.cpp:-1: first defined here

**Comment by afterburn420** [comment]
I commented out this code in moc_avplayer.cpp and live channels work perfectly  now -->

// SIGNAL 6
//void QtAV::AVPlayer::mediaStatusChanged(QtAV::MediaStatus _t1)
//{
//    void *_a[] = { Q_NULLPTR, const_cast<void*>(reinterpret_cast<const void*>(&_t1)) };
//    QMetaObject::activate(this, &staticMetaObject, 6, _a);
//}

Thanks for the fix - your code update works great.

Now (and before) the only bug I find is if you pause, ff, or rewind on videoclub movies it freezes up.

In other words it never starts playing again if you stop the video for any reason in videoclub.

**Comment by mvasilchuk** [comment]
I fixed that last bug, and also fixed play/pause action.  
BTW, in the new version I've added aspect ratio submenu into profiles menu to be able to change video aspect ratio.

**Comment by afterburn420** [comment]
Great - I will give it a try and let you know how it works for me...thanks

**Comment by afterburn420** [comment]
It worked great when I first tried it.

Due to my own fault (plugged in a bad hdmi cable) I froze my computer.

After a restart yasem always crashes with a segmentation fault in qt5core.dll.

Where does yasem store the program data (setup and such) - I believe I have corrupt it.

**Comment by afterburn420** [comment]
I went to c:/users/<myusername>/appdata/roaming and deleted the yasem folder there.

I am still getting segfault errors every time I try to start yasem (even older known good files).

It looks like it is not in the yasem config file - so I will have to find the error on my computer.

Unless there is other stored data that may be causing it.

**Comment by afterburn420** [comment]
here is the crash data in case it helps (if it is even a yasem problem)-->

 Problem Event Name:	APPCRASH
  Application Name:	yasem.exe
  Application Version:	0.1.0.0
  Application Timestamp:	002e9828
  Fault Module Name:	Qt5Core.dll
  Fault Module Version:	5.4.0.0
  Fault Module Timestamp:	777a675f
  Exception Code:	c0000005
  Exception Offset:	00071b00
  OS Version:	6.1.7601.2.1.0.256.1
  Locale ID:	1033
  Additional Information 1:	0a9e
  Additional Information 2:	0a9e372d3b4ad19135b953a78882e789
  Additional Information 3:	0a9e
  Additional Information 4:	0a9e372d3b4ad19135b953a78882e789

**Comment by afterburn420** [comment]
this is the call stack--> >	Qt5Core.dll!688f1b00()

**Comment by mvasilchuk** [comment]
Sorry, but this information is useless. Could you please run yasem in Debug mode from QtCreator? (change version to Debug and press Debug button)

**Comment by afterburn420** [comment]
here is an image of the debug screen after crash-->

![yasem-debug](https://cloud.githubusercontent.com/assets/10368178/6097754/368d29ea-af97-11e4-8b88-0f325575ab00.png)

if you need more then just let me know what you need.

**Comment by afterburn420** [comment]
Unless it is something that yasem saved causing it then I am sure that it is not in your code.

After my computer crashed I can no longer run any version of yasem - even the ones from earlier that were running before.

This computer has been running for a couple of years on this install - and has recently been getting issues with file ownership and permissions.

I believe that the crash opened up a new permission issue that is causing the problem.

My computer needs a fresh install of the OS - It just has not gotten it due to having to migrate the huge amount of data stored on it (1Tb drive filled~75%)

**Comment by mvasilchuk** [comment]
Try to do next:
1. In you GIT tool or git console run "git reset --soft" (or even --hard, but all your changes will be lost)
2. In QtCreator run "Run qmake" and  "Rebuild"
3. In yasem config directory check that in profiles/web-gui-config.ini value "url" is starting either from "file:///C:/whatever" (full path) or contains relative path to GUI subdir or compiled binary (i.e. doesn't start from / or C:/whatever).
4. Run the app under debugger to see is it works

**Comment by afterburn420** [comment]
contents of web-gui-config.ini -->

[profile]
uuid=web-gui-config
name=web-gui-config
classid=web-gui-plugin

[common]
url=gui/html/main.html

**Comment by afterburn420** [comment]
I just tried one of the old versions that used to work - it crashes with this error -->

[FATAL] (global\qglobal.cpp:2810 ASSERT: "browser != NULL" in file c:\yasem\yasem\plugins\yasem-web-server\serverthread.cpp' line 61
[FIXME] printCallStack() is only supported under linux

**Comment by afterburn420** [comment]
Something odd that I noticed - if I start yasem, let it crash and leave it open then I can start another instance of yasem and it runs fine

**Comment by mvasilchuk** [comment]
Haven't you changed Qt version? The bug may happens if you have DLLs from different Qt version in yor bin/ folder. Try to remove that folder and rebuild the project if nothing helps.

**Comment by afterburn420** [comment]
I could kick myself.

I have two versions of qtcreator installed - one to qt directory and the other to qt2 directory.

Only the one in qt2 directory works properly - so I always use it to compile with.

The other day I accidentally opened the version that does not work properly.

I had shut it down right away and opened the good version promptly forgetting about it.

when I had opened the faulty version it obviously (now) changed my qt path variables - even though the good version still showed its path. (in manage kits)

I ran my qt.bat file that I had copied and changed the paths to the good version - and now everything is working the way that it is supposed to work. (the qt.bat file automatically sets path variables and then starts qtcreator)

I will now change my qt2 shortcut to point to the qt.bat file for startup instead of the exe file to make sure that I never have this problem in the future.

Thank you for pointing me in the right direction - you can kick me too :)

**Comment by afterburn420** [comment]
Everything is running great now - and your new additions are working well too..they are not causing bugs etc.

Yasem is working almost as good as my actual mag receiver.
